### PR TITLE
[consensus] remove ChainedBftSMR

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

The only responsibility of ChainedBftSMR today is to spawn the event loop and
hold the runtime, so we can just return the runtime as other services.

All the tests written in chained_bft_smr_test should be either move to
event_processor_test to test the event driven response or smoke tests as
integration tests.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Existing tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

**Comment by dmitri-perelman** [comment]
Oh wow, seeing all the chained bft smr tests removed hurts...
So the chained_bft_smr_tests today are using network playground to control the message delivery and thus play different scenarios. This is not something we can easily do with the smoke tests today.
I guess it's a matter of terminology, but the smoke tests today are using real DB / execution / network / state-sync.
Chained-bft-smr tests could mock them all and test consensus functionality only. Isn't it a good thing to have?

**Comment by zekun000** [comment]
Totally agree we don't have the ability to control network in smoke test which should be added.
There's two types of tests there 
1. test expected behavior upon certain event which belongs to unit test in event_processor
2. test overall system works (with fault tolerance) which belongs to e2e tests in smoke tests

Here's a list of mapping of the existing tests:
Already exist:
basic_start_test  -> smoke_test_multi_node
start_with_proposal_test -> smoke_test_single_node
happy_path_with_multi_proposer -> smoke_test_multi_node with multi proposer 
basic_full_round_test -> smoke_test_multi_node
basic_block_retrieval -> event_processor::process_block_retrieval
basic_state_sync -> test_basic_state_synchronization
basic_commit_and_restart -> test_basic_restartability
basic_commit_and_restart_from_clean_storage -> test_startup_sync_state_with_empty_consensus_db
chain_with_nil_blocks -> event_processor::nil_vote_on_timeout
aggregate_timeout_votes -> event_processor::process_vote_timeout_msg_test

Should add:
state_sync_on_timeout -> event_processor::sync_on_timeout
block_retrieval_with_timeout -> event_processor::block_retrieval_on_timeout
secondary_proposers -> event_processor::secondary_vote_on_timeout 
sync_info_sent_if_remote_stale -> event_processor::sync_info_sent_if_remote_stale

File: consensus/src/consensus_provider.rs [hunk 1]
```diff
@@ -46,23 +37,44 @@ pub fn make_consensus_provider(
...
+        .expect("Failed to create Tokio runtime!");
     let storage = Arc::new(StorageWriteProxy::new(node_config, libra_db));
     let txn_manager = Box::new(MempoolProxy::new(consensus_to_mempool_sender));
     let state_computer = Arc::new(ExecutionProxy::new(executor, state_sync_client));
-    let input = ChainedBftSMRInput {
-        network_sender,
-        network_events,
-        safety_rules_manager: SafetyRulesManager::new(node_config),
-        state_computer,
+    let safety_rules_manager = SafetyRulesManager::new(node_config);
```
[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/src/consensus_provider.rs:50]: it's just copied over, makes sense to let constructor do more work

**Comment by zekun000** [comment]
yup, going to do some cleanup and add missing event_processor test

**Comment by ninjaahhh** [comment]
ah I will miss `chained_bft_smr_test.rs`

**Pull request closed without merging**